bin/program.py

```diff
@@ -403,20 +407,21 @@ def build(self, bar):
         if self.tmpdir.is_file():
             self.tmpdir.unlink()
         self.tmpdir.mkdir(parents=True, exist_ok=True)
-        self.timestamp = 0
         self.input_files = []
+        hashes = []
         for f in self.source_files:
```

Comment:
maybe ensure this is sorted so that `combine_hashes` is deterministic

Reply:
combine_hashes sorts the given list